src/features/earn/providers/yearn/api/api.ts

```diff
@@ -163,19 +187,23 @@ export class YearnVaultApi {
   }
 
   async allowance(input: Allowanceinput): Promise<string> {
-    const { userAddress, spenderAddress, tokenContractAddress } = input
+    const { userAddress, tokenContractAddress } = input
     const depositTokenContract: any = new this.web3.eth.Contract(erc20Abi, tokenContractAddress)
-    return depositTokenContract.methods.allowance(userAddress, spenderAddress).call()
+    return depositTokenContract.methods.allowance(userAddress, ssRouterContractAddress).call()
   }
 
   async depositEstimatedGas(input: TxEstimatedGasInput): Promise<BigNumber> {
-    const { amountDesired, userAddress, vaultAddress } = input
-    const vaultContract = new this.web3.eth.Contract(yv2VaultAbi, vaultAddress)
-    const estimatedGas = await vaultContract.methods
-      .deposit(amountDesired.toString(), userAddress)
+    const { amountDesired, userAddress, tokenContractAddress, vaultAddress } = input
+
+    const routerContract = new this.web3.eth.Contract(ssRouterAbi, ssRouterContractAddress)
```

Comment:
Is there any reason not to instantiate the router contract in the constructor and store it vs creating a `new` instance for each function call?

Reply:
done.